bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/protocols/AbstractScalarWebProtocol.java

```diff
@@ -0,0 +1,1113 @@
+                                    final double dmin = min == null || min.isInfinite() || min.isNaN() ? 0
+                                            : min.doubleValue();
+                                    final double dmax = max == null || max.isInfinite() || max.isNaN() ? 100
+                                            : max.doubleValue();
+                                    final double dstep = step == null || step.isInfinite() || step.isNaN() ? 1
+                                            : step.doubleValue();
+                                    for (double p = dmin; p <= dmax; p += (dstep == 0 ? 1 : dstep)) {
+                                        final String opt = Double.toString(p);
+                                        options.add(new StateOption(opt, opt));
+                                    }
+                                }
+                                if (min != null) {
+                                    bld = bld.withMinimum(new BigDecimal(min));
+                                }
+                                if (max != null) {
+                                    bld = bld.withMaximum(new BigDecimal(max));
+                                }
+                                if (step != null) {
+                                    bld = bld.withStep(new BigDecimal(step));
+                                }
+                                if (!options.isEmpty()) {
+                                    bld = bld.withOptions(options);
+                                }
+                            }
+                        }
+
+                        break;
+                    case GeneralSetting.INTEGERTARGET:
+                        if (set.isUiSlider()) {
+                            descriptors.add(createDescriptor(channel, "Dimmer", "scalargeneralsettingdimmer",
+                                    prefix + " " + label, prefix + " for " + label));
+
+                        } else {
+                            descriptors.add(createDescriptor(channel, "Number", "scalargeneralsettingnumber",
+                                    prefix + " " + label, prefix + " for " + label));
+                        }
+
+                        if (candidates.size() > 0) {
+                            final GeneralSettingsCandidate candidate = candidates.get(0);
+
+                            final Double min = candidate.getMin(), max = candidate.getMax(), step = candidate.getStep();
+                            if (min != null || max != null || step != null) {
+                                final List<StateOption> options = new ArrayList<>();
+                                if (set.isUiPicker()) {
+                                    final int imin = min == null ? 0 : min.intValue();
+                                    final int imax = max == null ? 100 : max.intValue();
+                                    final int istep = step == null ? 1 : step.intValue();
```

Comment:
You are doing a double to int conversion yet you aren't checking for Nan and Infinity like the `case DOUBLETARGET:` does above.